ndb/src/google/cloud/ndb/eventloop.py

```diff
@@ -30,16 +31,228 @@
 ]
 
 
-def add_idle(*args, **kwargs):
-    raise NotImplementedError
+def _noop(*args, **kw):
+    """Do nothing."""
+
+# TODO: Use utils.logging_debug when implemented
+_logging_debug = _noop
+
+
+_Event = collections.namedtuple('_Event', (
+    'when', 'callback', 'args', 'kwargs'))
+
+
+class _Clock(object):
+    """A clock to determine the current time, in seconds."""
+
+    def now(self):
+        """Returns the number of seconds since epoch."""
+        return time.time()
+
+    def sleep(self, seconds):
+        """Sleeps for the desired number of seconds."""
+        time.sleep(seconds)
 
 
 class EventLoop:
-    __slots__ = ()
+    """Constructor.
+
+    Args:
+      clock: an eventloop._Clock object. Defaults to a time-based clock.
```

Comment:
Can we find out if any "real" callers pass in a clock (outside our own tests)?

Reply:
None in NDB, and I don't think "real code" (TM) is instantiating EventLoop directly.  This commit is clear that it's there for testing: b3d4925d6f34fcc1c05742d00e2e3527664a75c9#diff-07e4b05699808712834ec5c483a21849